annotation-processor/src/main/java/org/hibernate/validator/ap/checks/annotationparameters/AnnotationParametersAbstractCheck.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.checks.annotationparameters;
+
+import org.hibernate.validator.ap.checks.AbstractConstraintCheck;
+import org.hibernate.validator.ap.checks.ConstraintCheckError;
+import org.hibernate.validator.ap.util.AnnotationApiHelper;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Checks, that parameters used on annotation are valid.
+ */
+public abstract class AnnotationParametersAbstractCheck extends AbstractConstraintCheck {
+
+	protected final AnnotationApiHelper annotationApiHelper;
+
+	private Set<String> annotationClasses;
+
+	public AnnotationParametersAbstractCheck(AnnotationApiHelper annotationApiHelper, String... annotationClass) {
+		this.annotationApiHelper = annotationApiHelper;
+		this.annotationClasses = new HashSet<>( Arrays.asList( annotationClass ) );
+
+	}
+
+	@Override
+	public Set<ConstraintCheckError> checkField(VariableElement element, AnnotationMirror annotation) {
+
```

Comment:
FTR, the empty line was a style we (I) used at some point, but nowadays we don't do it anymore.